web/src/components/CardInspectDialog.tsx

```diff
@@ -0,0 +1,403 @@
+import { formatDateForDisplay } from '@/lib/utils';
+
+import ReviewHistory from './ReviewHistory';
+
+interface CardInspectDialogProps {
+    card: CardOut;
+    trigger?: React.ReactNode;
+    open?: boolean;
+    onOpenChange?: (open: boolean) => void;
+    onUpdateSuccess?: () => void;
+    onDeleteSuccess?: () => void;
+    onUpdateError?: (error: string) => void;
+    onDeleteError?: (error: string) => void;
+    onNext?: () => void;
+    onPrev?: () => void;
+    hasNext?: boolean;
+    hasPrev?: boolean;
+}
+
+export default function CardInspectDialog({
+    card,
+    trigger,
+    open,
+    onOpenChange,
+    onUpdateSuccess,
+    onDeleteSuccess,
+    onUpdateError,
+    onDeleteError,
+    onNext,
+    onPrev,
+    hasNext,
+    hasPrev,
+}: CardInspectDialogProps) {
+    usePageShortcuts('decks');
+    const { registerAction, unregisterAction } = useShortcutActions();
+
+    const [internalOpen, setInternalOpen] = useState(false);
+    const isOpen = open ?? internalOpen;
+    const setIsOpen = onOpenChange ?? setInternalOpen;
+
+    const { data: decks } = useQuery({
+        queryKey: ['decks'],
+        queryFn: () => getDecksDecksGet(),
+    });
+
+    const {
+        data: reviewHistory,
+        isLoading: reviewHistoryLoading,
+        isError: reviewHistoryError,
+    } = useQuery({
+        queryKey: ['reviews', card.id],
+        queryFn: () =>
+            getReviewHistoryReviewsCardIdGet({ path: { card_id: card.id } }),
+        staleTime: 5 * 60 * 1000,
+    });
+
+    const deleteCardMutation = useMutation({
+        mutationFn: () =>
+            deleteCardCardsCardIdDelete({ path: { card_id: card.id } }),
+        onSuccess: () => {
+            cardForm.reset();
+            setIsOpen(false);
+            onDeleteSuccess?.();
+            // TODO toast?
+        },
+        onError: (err: unknown) => {
+            const errorMessage = `There was an error deleting card: ${(err as Error)?.message ?? 'no details found'}`;
+            onDeleteError?.(errorMessage);
+        },
+    });
+
+    const updateCardMutation = useMutation({
+        mutationFn: (values: z.infer<typeof cardFormSchema>) =>
+            updateCardCardsCardIdPatch({
+                path: { card_id: card.id },
+                body: values,
+            }),
+        onSuccess: () => {
+            cardForm.reset({
+                content: card.content,
+                deck_id: card.deck_id,
+            });
+            setIsOpen(false);
+            onUpdateSuccess?.();
+            // TODO toast?
+        },
+        onError: (err: unknown) => {
+            const errorMessage = `There was an error updating card: ${(err as Error)?.message ?? 'no details found'}`;
+            onUpdateError?.(errorMessage);
+        },
+    });
+
+    const cardFormSchema = z.object({
+        content: z.string().min(1, 'Card must have contents'),
+        deck_id: z.string().min(1, 'Card must have a parent deck'),
+    });
```

Comment:
I think this should ensure that the schema satisfies the type like for card creation, just CardUpdate instead of CardCreate.

```
const cardFormSchema = z.object({
        content: z.string().min(1, 'Card must have contents'),
        deck_id: z.string().min(1, 'Card must have a parent deck'),
    }) satisfies z.ZodType<CardCreate>;
```